### ISSUE TITLE
Defense tooltip could handle limit better

### ISSUE THREAD

**Issue by sevu** [wesnoth/wesnoth#5320]
In this tooltip, shown when mousing over the defense value, it shows the defense of both terrains together with "min". However, in this case there's a defense limit on forest.

How could this be handled? Showing max for forest? Or not showing min/max (looks already a bit strange).

![defense_tooltip](https://user-images.githubusercontent.com/21158813/100481601-dba35500-30f4-11eb-846d-6dbb5e9607d2.png)

**Comment by Pentarctagon** [comment]
Yeah, the "min" parts looks weird, and I'm not really sure how it's useful - what really matters is the resulting defense rather than how it decided what that defense is.

**Comment by stevecotton** [comment]
I think it would make more sense to put the min/max on a different line, so

```
Terrain: Great Tree
    Minimum of:
    Flat: 40%
    Forest: 30% (cap)
Defense: 30%
```

The current implementation in src/reports.cpp's unit_defense() is written to handle complex mixed terrains with defense such as max(a, min(b, c)), but those not only aren't used in the current terrains, they aren't supported in the engine (#4279).

**Comment by CelticMinstrel** [comment]
> what really matters is the resulting defense rather than how it decided what that defense is.

The breakdown of how it was decided seems to be the entire point of that tooltip. If you just want the final defense value, you don't even need the tooltip.

> those not only aren't used in the current terrains, they aren't supported in the engine

But they might be supported in the engine in the future, depending on what's decided for #4279, so until that's decided we should operate under the assumption that they could be supported.
